### ISSUE TITLE
Add mode option to redirect to non-API - direct S3 URL

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandidav#92]
To overcome https://savannah.nongnu.org/bugs/index.php?65376 as investigated in
- https://github.com/dandi/dandisets-healthstatus/issues/66#issuecomment-1966658830
For use under FUSE it would not matter and actually would be more performant to avoid 2nd redirect.

I wonder if it could somehow be "run time option", i.e. is it possible to provide options to webdav instance e.g. have
https://dandi.centerforopenneuroscience.org/dandisets/000003/?redirect=dandi-api or https://dandi.centerforopenneuroscience.org/dandisets/000003/?redirect=s3 and then use those options during FUSE mount so we could have a single instance but behavior would be changed by client?

**Comment by jwodder** [comment]
@yarikoptic

> I wonder if it could somehow be "run time option"

This would rely entirely on exactly how davfs2, webdavfs, or any other WebDAV-FUSEr constructs URLs for files & directories within the mount.  The URL spec seems to require that the query portion of a base URL be discarded when constructing a descendant URL, but I wouldn't count on either client we're using to honor that properly, and they could start honoring it at any future point in time anyway.

Theoretically, if `dandidav` were to preserve query parameters in `PROPFIND` requests by including them in `href`s in `PROPFIND` responses, a properly-paranoid client would end up using URLs with query parameters when `GET`ting files, but that would only work here if the client either (a) caches `href`s (and I'm not sure if either client does) or (b) does a recursive `PROPFIND` from the root of the server on every file operation (which would be incredibly inefficient).

In summary, I'd rather not try to get this working as a "run time option" query parameter and would rather it be a CLI option.

**Comment by yarikoptic** [comment]
ok.  we might be doomed to run multiple instances then -- something to keep in mind while preparing official deployment.
